candle-core/src/backprop.rs

```diff
@@ -707,6 +707,9 @@ impl Tensor {
                         let sum_grad = grads.or_insert(arg)?;
                         *sum_grad = sum_grad.add(&arg_grad)?
                     }
+                    Op::Unary(_arg, UnaryOp::Sign) => {
+                        Err(Error::BackwardNotSupported { op: "sign" })?
```

Comment:
The backward is actually easy to implement here, we should just do nothing as there is no gradient flowing back on such a discrete operator.

Reply:
would that simply be the following:

```
let sum_grad = grads.or_insert(arg)?;
                        let arg_grad = grad.to_device(sum_grad.device())?;
                        *sum_grad = sum_grad.add(&arg_grad)?
```

I'm not yet quite familiar with internals of the backpropagation tracking

Reply:
Nope, literally nothing should be done, and also around line 110 in the same file handle it as `Round` etc. The gradient doesn't flow through operator that are not contiguous (it's not necessary for correctness but will make it more efficient).

Reply:
Ah I see. Okay I've updated round and sign to be no-ops now

Reply:
Ah that's not actually what I meant, let me actually do the tweak (it's fine to do the change you did on `Round` but a bit unrelated to this PR). Also note that it's the other way around compared to your commit message, we prevent the gradient from flowing through these ops as they are not continuous.